aws/ecs/packer/to-upload/peers.sh

```diff
@@ -1,30 +1,235 @@
+        local __possible_spot_fleet="$(${__awscli} ec2 describe-spot-fleet-instances --spot-fleet-request-id "${spot_fleet_id}" --query "${__spot_fleet_instance_filter}" --output text)"
+        if [ "${__possible_spot_fleet}" == "${__instance_id}" ]; then
+          __spot_fleet_request_id="${spot_fleet_id}"
+          break
+        fi
+      done
+      if ! check_data "${__spot_fleet_request_id}"; then
+        >&2 echo "${FUNCNAME[0]} WARNING Spot Fleet ID not found.. must not be in a spot fleet"
+        return 1
+      fi
+    else
+      return 1
+    fi
+  fi
+}
+
+get_peer_spot_fleet_instance_ids() {
+  if ! check_data "${__peer_spot_fleet_instance_ids}"; then
+    get_awscli
+    if get_spot_fleet_request_id; then
+      get_instance_id
+      local __peer_instances_filter="ActiveInstances[?InstanceId != '${__instance_id}'].InstanceId"
+      __peer_spot_fleet_instance_ids="$(${__awscli} ec2 describe-spot-fleet-instances --spot-fleet-request-id "${__spot_fleet_request_id}" --query "${__peer_instances_filter}" --output text)"
+      check_exit $?
+      if ! check_data "${__peer_spot_fleet_instance_ids}"; then
+        >&2 echo "${FUNCNAME[0]} WARNING No other instances found in spot fleet: ${__spot_fleet_request_id}"
+        return 1
+      fi
+    else
+      return 1
+    fi
+  fi
+}
+
+# IF instance has a KEY ${WEAVE_AWS_PEER_TAG} find all the others like it
+#
+# i.e. if this instance is tagged WEAVE_ECS=prod this will get all the other instance ids that have that key/value pair
+get_peer_tag_instance_ids() {
+  if ! check_data "${__peer_tag_instance_ids}"; then
+    get_instance_id
+    get_awscli
+    if check_data "${WEAVE_AWS_PEER_TAG}"; then
+      local __instance_tag_filter="Reservations[].Instances[?InstanceId == '${__instance_id}'].Tags[][?Key == '${WEAVE_AWS_PEER_TAG}'].Value"
+      local __keyval=
+      __keyval=$(${__awscli} ec2 describe-instances --filters "Name=tag-key,Values=${WEAVE_AWS_PEER_TAG}" --query "${__instance_tag_filter}" --output text)
+      if check_data "${__keyval}"; then
+        local __instance_ids_filter="Name=tag-key,Values=${WEAVE_AWS_PEER_TAG},Name=tag-value,Values=${__keyval}"
+        local __instance_ids_query="Reservations[].Instances[?InstanceId != '${__instance_id}'].InstanceId"
+        __peer_tag_instance_ids=$(${__awscli} ec2 describe-instances --filters "${__instance_ids_filter}" --query "${__instance_ids_query}" --output text)
+        if ! check_data "${__peer_tag_instance_ids}"; then
+          >&2 echo "${FUNCNAME[0]} WARNING No other instances found with tag ${WEAVE_AWS_PEER_TAG}=${__keyval}"
+          return 1
+        fi
+      else
+        return 1
+      fi
+    else
+      >&2 echo "${FUNCNAME[0]} WARNING WEAVE_AWS_PEER_TAG not set in /etc/weave/weave.config"
+      return 1
+    fi
+  fi
+}
+
+get_peer_private_ips() {
+  if ! check_data "${__peer_private_ips}"; then
+    get_awscli
+    local __instance_ids=
+    if get_peer_tag_instance_ids; then
```

Comment:
I didn't really consider it a big deal b/c the peers.sh script shouldn't be sourced anywhere, and that pattern is IMHO more difficult to understand, and makes the order of the peer instance_id collection less obvious